DSharpPlus/Utilities.cs

```diff
@@ -322,6 +322,32 @@ internal static void LogTaskFault(this Task task, ILogger logger, LogLevel level
         value = kvp.Value;
     }
 
+    /// <summary>
+    /// Creates a snowflake from a given <see cref="DateTimeOffset"/>. This can be used to provide "timestamps" for methods
+    /// like <see cref="DiscordChannel.GetMessagesAfterAsync"/>.
+    /// </summary>
+    /// <param name="dateTimeOffset">DateTimeOffset to create a snowflake from.</param>
+    /// <returns>Returns a snowflake representing the given date and time.</returns>
+    public static ulong CreateSnowflake(DateTimeOffset dateTimeOffset)
+    {
+        var diff = dateTimeOffset - DiscordClient._discordEpoch;
+        return (ulong)diff.TotalMilliseconds << 22;
+    }
+    
+    /// <summary>
+    /// Creates a snowflake from a given <see cref="DateTime"/>.
+    /// This can be used to provide "timestamps" for methods
+    /// like <see cref="DiscordChannel.GetMessagesAfterAsync"/>.
+    /// </summary>
+    /// <param name="dateTimeOffset">DateTimeOffset to create a snowflake from.</param>
+    /// <returns>Returns a snowflake representing the given date and time.</returns>
+    /// <remarks>THIS METHOD ASSUMES THAT THE DATETIME IS IN UTC.</remarks>
```

Comment:
this should be a sign to not create this method - we don't support DateTime across the library for this reason